deploy-service/common/src/main/java/com/pinterest/deployservice/handler/CommonHandler.java

```diff
@@ -152,33 +156,50 @@ String generateMessage(String buildId, EnvironBean envBean, DeployState state, D
         String webLink = deployBoardUrlPrefix + String.format("/env/%s/%s/deploy/",
             envBean.getEnv_name(),
             envBean.getStage_name());
+        
+        BuildTagsManager buildTagsManager = new BuildTagsManagerImpl(this.tagDAO);
+        TagBean tagBean = buildTagsManager.getEffectiveBuildTag(buildBean);
 
         String action = getDeployAction(deployType);
         if (state == DeployState.SUCCEEDING) {
             // TODO this is Slack specific, screw hipchat for now
-            return String.format("%s/%s: %s %s/%s completed successfully. See details <%s>",
+
+            if (tagBean != null && tagBean.getValue() == TagValue.BAD_BUILD) {
+                return String.format("WARNING: %s/%s: %s %s/%s completed successfully, but running on bad build. See details <%s>",
                 envBean.getEnv_name(),
                 envBean.getStage_name(),
                 action,
                 buildBean.getScm_branch(),
                 buildBean.getScm_commit_7(),
                 webLink);
+            } else {
+                return String.format("%s/%s: %s %s/%s completed successfully. See details <%s>",
+                envBean.getEnv_name(),
+                envBean.getStage_name(),
+                action,
+                buildBean.getScm_branch(),
+                buildBean.getScm_commit_7(),
+                webLink);
+            }
```

Comment:
Can we use different templates so the code is not duplicated?

Reply:
do you mean changing the strings

Reply:
Like
```java
String template = (tagBean != null && tagBean.getValue() == TagValue.BAD_BUILD) ?
"WARNING: %s/%s: %s %s/%s completed successfully, but running on bad build. See details <%s>" :
"%s/%s: %s %s/%s completed successfully. See details <%s>";
```